otx/algorithms/anomaly/adapters/anomalib/callbacks/inference.py

```diff
@@ -44,54 +47,123 @@ def __init__(self, otx_dataset: DatasetEntity, labels: List[LabelEntity], task_t
         # TODO; refactor Ignore too many locals
         # pylint: disable=too-many-locals
         outputs = outputs[0]
-        pred_scores = np.hstack([output["pred_scores"].cpu() for output in outputs])
-        pred_labels = np.hstack([output["pred_labels"].cpu() for output in outputs])
-        anomaly_maps = np.vstack([output["anomaly_maps"].cpu() for output in outputs])
-        pred_masks = np.vstack([output["pred_masks"].cpu() for output in outputs])
-
-        # Loop over dataset again to assign predictions
-        for dataset_item, pred_score, pred_label, anomaly_map, pred_mask in zip(
-            self.otx_dataset, pred_scores, pred_labels, anomaly_maps, pred_masks
-        ):
-            probability = pred_score if pred_label else 1 - pred_score
-            if self.task_type == TaskType.ANOMALY_CLASSIFICATION:
-                label = self.anomalous_label if pred_label else self.normal_label
-            if self.task_type == TaskType.ANOMALY_DETECTION:
-                annotations = create_detection_annotation_from_anomaly_heatmap(
-                    hard_prediction=pred_mask,
-                    soft_prediction=anomaly_map,
-                    label_map=self.label_map,
-                )
-                dataset_item.append_annotations(annotations)
-                label = self.normal_label if len(annotations) == 0 else self.anomalous_label
-            elif self.task_type == TaskType.ANOMALY_SEGMENTATION:
-                annotations = create_annotation_from_segmentation_map(
-                    hard_prediction=pred_mask.squeeze().astype(np.uint8),
-                    soft_prediction=anomaly_map.squeeze(),
-                    label_map=self.label_map,
-                )
-                dataset_item.append_annotations(annotations)
-                label = self.normal_label if len(annotations) == 0 else self.anomalous_label
+        # collect generic predictions
+        pred_scores = torch.hstack([output["pred_scores"].cpu() for output in outputs])
+        pred_labels = torch.hstack([output["pred_labels"].cpu() for output in outputs])
+        anomaly_maps = torch.vstack([output["anomaly_maps"].cpu() for output in outputs])
+        pred_masks = torch.vstack([output["pred_masks"].cpu() for output in outputs])
 
-            dataset_item.append_labels([ScoredLabel(label=label, probability=float(probability))])
+        # add the predictions to the dataset item depending on the task type
+        if self.task_type == TaskType.ANOMALY_CLASSIFICATION:
+            self._process_classification_predictions(pred_labels, pred_scores)
+        elif self.task_type == TaskType.ANOMALY_DETECTION:
+            # collect detection predictions
+            pred_boxes = []
+            box_scores = []
+            box_labels = []
+            for output in outputs:
+                pred_boxes.extend(output["pred_boxes"])
+                box_scores.extend(output["box_scores"])
+                box_labels.extend(output["box_labels"])
```

Comment:
![image](https://user-images.githubusercontent.com/17574420/217817540-cdeda624-a522-4e83-a46d-c31d33a55e4b.png)
Maybe we could use list comprehension here, which is shorter and more efficient.

Reply:
```suggestion
            pred_boxes = [output["pred_boxes"] for output in outputs]
            box_scores = [output["box_scores"] for output in outputs]
            box_labels = [output["box_labels"] for output in outputs]
```

Reply:
This won't work because we have to flatten the list of lists. We could do the following:
```suggestion
            pred_boxes = [box for output in outputs for box in output["pred_boxes"]]
            box_scores = [score for output in outputs for score in output["box_scores"]]
            box_labels = [label for output in outputs for label in output["box_labels"]]
```
but I feel it would make the code less readable

Reply:
oh I see. Yeah, then ignore this comment. I'll resolve it